dev/plugins/hu.elte.txtuml.diagnostics/src/hu/elte/txtuml/diagnostics/session/DiagnosticsPlugin.java

```diff
@@ -41,11 +45,28 @@ public DiagnosticsPlugin(int diagnosticsPort, String projectName, String working
 		modelMapper = new ModelMapper(projectName);
 		instanceRegister = new InstanceRegister();
 		animator = new Animator(instanceRegister, modelMapper);
+		server = new DiagnosticsServer(this);
+		
+		try {
+			server.start(httpPort);
+		} catch (IOException e) {
+			Logger.sys.error("Couldn't start HTTP server on port " + httpPort + " in service instance 0x", e);
+			return;
+		}
+		
 		thread = new Thread(this, "txtUMLDiagnosticsPlugin");
 		thread.start();
 		//Logger.logInfo("txtUML DiagnosticsPlugin started"));
 	}
 	
+	public void setDelay(int speed){
+		DELAY = new AtomicInteger(speed);
```

Comment:
This assignment is not thread safe, use `AtomicInteger.set(int)` instead. Also change `speed` to `delay`.

Reply:
My request above still holds.